src/lib/utils/isCloseToBottom.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Detects if a <ScrollView /> has scrolled to the bottom.
```

Comment:
Is there a util anywhere in Emission that does something similar?

Reply:
`<ListView />` has end reached detection built in but not for `<ScrollView />`, let's use that one.

Reply:
@l2succes - It looks like `ListView` needs a bunch of setup, like 

```javascript
this.state = {
  dataSource: new ListView.DataSource({
    rowHasChanged: (row1, row2) => row1 !== row2,
  }).cloneWithRows(this.props.shows),
}
```
And then you have to pass in data to render individual rows. Do you think thats more straightforward than this util? I'm thinking that it could be useful since a `ScrollView` is simpler and can render `children`. What did you have in mind exactly?

Reply:
@damassi Sorry for the confusion, I meant the util makes sense. Let's use it as I doubt we already have one in emission

Reply:
👍

Reply:
There’s `FlatList` which doesn’t require the data source: https://facebook.github.io/react-native/docs/flatlist.html

I’d suggest using builtin components where possible, as we get optimisations that FB continuously does for free.

For the masonry grid where there’s no builtin component that we could leverage, we lifted the end reached code from FB’s implementations https://github.com/artsy/emission/blob/9ac120c2195767d4e494e69712dd196857eb6350/src/lib/Components/ArtworkGrids/InfiniteScrollGrid.tsx#L261-L274